### PR TITLE
Fix Intrinsic layout in IE11

### PR DESCRIPTION
**Pull request by jridgewell**
IE11 has a bug with properly scaling SVG images, which breaks `intrinsic` layout. Intrinsic uses a special SVG `data:` URI `` to perform responsive resizing.

What's actually important about this is that it's an `` tag, which has "intrinsic" sizing and won't collapse when the parent is floated. Using SVG allows us to simulate the aspect ratio of the parent container, and it's very tiny to use. But if we could instead generate a image with the correct aspect ratio on the fly, we could use that value, too.

Surprisingly, `<canvas>` [support is pretty good](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL) in IE. We can use a canvas to generate a `data:` URI with the correct size. And the code to generate it is tiny, even if the URIs it generates are extremely large (several kb). So we definitely don't want to do this for every browser.

See a generated demonstration at https://output.jsbin.com/ticifer/quiet, and the generation code running at https://output.jsbin.com/muqeji/quiet.

Fixes https://github.com/ampproject/amphtml/issues/28398.

### CODE REVIEW

File: src/layout.js [hunk 1]
```diff
@@ -527,7 +529,12 @@ export function applyStaticLayout(element) {
     const intrinsicSizer = sizer.firstElementChild;
     intrinsicSizer.setAttribute(
       'src',
-      `data:image/svg+xml;charset=utf-8,<svg height="${height}" width="${width}" xmlns="http://www.w3.org/2000/svg" version="1.1"/>`
+      fixIeIntrinsic
```
Comment (kristoferbaxter) [inline src/layout.js:532]: Since this is specific to IE only, could we wrap the logic here with an `IS_ESM` check and not pass around the argument?

File: src/utils/img.js [hunk 2]
```diff
@@ -31,3 +31,21 @@ export function guaranteeSrcForSrcsetUnsupportedBrowsers(img) {
...
+ */
+export function transparentPng(width, height) {
+  const canvas = self.document.createElement('canvas');
+  canvas.width = width;
+  canvas.height = height;
+
+  // Canvases are fully transparent by default, so we don't actually need to
+  // draw anything.
+
+  return canvas.toDataURL();
```
Comment (kristoferbaxter) [inline src/utils/img.js:50]: Do we need to create a new canvas each time?

**Comment by samouri** [comment]
Note: I approved this because the code lgtm, but won't be able to test it manually until I get a vm for ie11 going.

File: src/utils/img.js (hunk 2 above)
Comment (kristoferbaxter) [reply src/utils/img.js:50]: Makes sense, thanks!

File: test/integration/test-amp-img.js [hunk 3]
```diff
@@ -136,21 +136,64 @@ describe
...
+            );
+          });
+        });
+      });
+
+    // IE can't scale SVG images, so intrinsic sizers use a PNG instead.
+    describe
+      .configure()
+      .enableIe()
+      .skipSafari()
```
Comment (choumx) [inline test/integration/test-amp-img.js:162]: If we run this particular test on multiple browsers, it doesn't really make sense to be nested under "Rendering of amp-img - Internet Explorer edge cases".

File: src/service/resources-impl.js [hunk 4]
```diff
@@ -34,6 +34,7 @@ import {
 import {getMode} from '../mode';
 import {getSourceUrl} from '../url';
 import {hasNextNodeInDocumentOrder, isIframed} from '../dom';
+import {checkAndFix as ieIntrinsicCheckAndFix} from './ie-intrinsic-bug';
```
[the remark this answers is not in the document]
Comment (jridgewell) [reply src/service/resources-impl.js]: Done.

File: src/utils/img.js [hunk 5]
```diff
@@ -31,3 +31,21 @@ export function guaranteeSrcForSrcsetUnsupportedBrowsers(img) {
...
+
+/**
+ * Generates a transparent PNG of a given width/height.
+ *
+ * @param {number} width
+ * @param {number} height
+ * @return {string}
+ */
+export function transparentPng(width, height) {
+  const canvas = self.document.createElement('canvas');
```
[the remark this answers is not in the document]
Comment (jridgewell) [reply src/utils/img.js]: Done.

File: test/integration/test-amp-img.js (hunk 3 above)
Comment (jridgewell) [reply test/integration/test-amp-img.js:162]: Renamed.

File: test/integration/test-amp-img.js [hunk 6]
```diff
@@ -136,21 +136,64 @@ describe
...
+          );
+          return imageLoadedPromise.then(() => {
+            const ampImg = fixture.doc.getElementById('intrinsic');
+            const width = parseInt(ampImg.getAttribute('width'), 10);
+            const height = parseInt(ampImg.getAttribute('height'), 10);
+            const bounds = ampImg.getBoundingClientRect();
+
+            expect(reduce(width, height)).to.deep.equal(
+              reduce(bounds.width, bounds.height)
+            );
```
[the remark this answers is not in the document]
Comment (jridgewell) [reply test/integration/test-amp-img.js:178]: I was worried about floating point issues. Updated.

File: src/service/ie-intrinsic-bug.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
```
Comment (choumx) [inline src/service/ie-intrinsic-bug.js:2]: 2020
